MdeModulePkg/Core/Dxe/Misc/MemoryProtectionSupport.c

```diff
@@ -1667,6 +1705,79 @@ SyncBitmapAndSetAccessAttributesInMemoryMap (
   }
 }
 
+/**
+  Removes the access attributes from memory map descriptors which match the elements in the
+  input IMAGE_PROPERTIES_RECORD list.
+
+  @param[in]      MemoryMapSize           A pointer to the size, in bytes, of the MemoryMap buffer
+  @param[out]     MemoryMap               A pointer to the buffer containing the memory map. This
+                                          memory map must be sorted.
+  @param[in]      DescriptorSize          Size, in bytes, of an individual EFI_MEMORY_DESCRIPTOR
+  @param[in]      NonProtectedImageList   List of IMAGE_PROPERTIES_RECORD entries. This list
+                                          must be sorted.
+
+  @retval   EFI_SUCCESS                   The bitmap was updated
+  @retval   EFI_INVALID_PARAMETER         MemoryMapSize was NULL, DescriptorSize was NULL,
+                                          Bitmap was NULL, or MemoryMap was NULL
+  @retval   EFI_NOT_FOUND                 No memory map entry matched an image properties record described
+                                          in the input IMAGE_PROPERTIES_RECORD list
+**/
+STATIC
+EFI_STATUS
+RemoveAttributesOfNonProtectedImageRanges (
+  IN CONST  UINTN                  *MemoryMapSize,
+  OUT       EFI_MEMORY_DESCRIPTOR  *MemoryMap,
+  IN CONST  UINTN                  *DescriptorSize,
+  IN        LIST_ENTRY             *NonProtectedImageList
+  )
+{
+  LIST_ENTRY               *NonProtectedImageRecordLink = NULL;
+  IMAGE_PROPERTIES_RECORD  *NonProtectedImageRecord     = NULL;
+  EFI_MEMORY_DESCRIPTOR    *MemoryMapEntry              = NULL;
+  EFI_MEMORY_DESCRIPTOR    *MemoryMapEnd                = NULL;
+
+  if ((MemoryMapSize == 0) || (MemoryMap == NULL) ||

```

Comment:
This is technically okay, but did you mean to check the value to be zero (in addition to the pointer being non-NULL)?

The edk2 style guideline prefers null checks to be against `NULL` (instead of `0`) if just checking the pointer.

Reply:
Was a mistake - fixed